### PR TITLE
refactor: write manifest file even if migration failed

### PR DESCRIPTION
**Pull request by lambda-0x**
fix: DOJ-116, https://github.com/dojoengine/dojo/issues/1218

### CODE REVIEW

**Comment by lambda-0x** [comment]
(codecov report seems misleading, have only renamed the struct for which its complaining about. i think it can be ignored)

**Comment by remybar** [comment]
Hi @lambda-0x !

I'm working on a PR related to writing `transaction_hash` and `block_number` in the manifest (see: https://github.com/dojoengine/dojo/pull/1651). As we get these data when the migration is done (`apply_diff`), the manifest file must be updated after the migration, at least for these data.

But I understand your point, so maybe we just have to update the manifest file 2 times: before and after `apply_diff`.

**Comment by lambda-0x** [comment]
Haha yes, I saw your PR after opening this. For your simplicity lets first merge your PR I will make the necessary changes required for this PR.

**Review by glihm — COMMENTED** [review]
Looks good man!
Some minor comments, and as discussed let's wait #1651 to be merged in order to ensure the global flow is ok in this one.

File: bin/sozo/src/ops/migration/mod.rs [hunk 1]
```diff
@@ -83,38 +84,52 @@ pub async fn execute(
...
+            update_manifests_and_abis(
+                ws,
+                local_manifest,
+                remote_manifest,
+                &manifest_dir,
+                MigrationOutput {
+                    world_address,
+                    world_tx_hash: None,
+                    world_block_number: None,
+                    success: false,
```
Comment (glihm) [inline bin/sozo/src/ops/migration/mod.rs:117]: Could we derive `Default` on `MigrationOutput` to have `..Default::default()` instead?

File: bin/sozo/src/ops/migration/mod.rs [hunk 2]
```diff
@@ -202,42 +217,47 @@ async fn update_manifest_abis(
...
-    if let Some(block_number) = migration_output.world_block_number {
-        ui.print(format!(
-            "\n🎉 Successfully migrated World on block #{} at address {}",
-            block_number,
-            bold_message(format!(
-                "{:#x}",
-                strategy.world_address().expect("world address must exist")
-            ))
-        ));
+    if migration_output.success {
```
Comment (glihm) [inline bin/sozo/src/ops/migration/mod.rs:239]: We may replace success with `partial` or `full`, to indicate that at least part of the migration was successful as the world is deployed.

**Review by glihm — COMMENTED** [review]
Great work man!

I have a situation:
If I modify the `actions.cairo` contract, the class hash changes. However, this change is not reflected into the `KATANA.toml` manifest if I kill katana during the base / world migration. However, new models are correctly reflected into the `KATANA.toml` even if the base migration fails.

Any idea?

**Review by glihm — APPROVED** [review]

**Pull request merged**